### PR TITLE
Updated gitignore files based on latest templates.

### PR DESCRIPTION
**Pull request by awhitford**
Updated `.gitignore` files based on latest templates.

### CODE REVIEW

File: example/.gitignore [hunk 1]
```diff
@@ -1,41 +1,162 @@
...
+.idea/tasks.xml
+.idea/gradle.xml
+.idea/assetWizardSettings.xml
+.idea/dictionaries
+.idea/libraries
+# Android Studio 3 in .gitignore file.
+.idea/caches
+.idea/modules.xml
+# Comment next line if keeping position of elements in Navigation Editor is relevant for you
+.idea/navEditor.xml
```
Comment (SirusCodes) [inline example/.gitignore:56]: these all are equivalent as .idea/ ??

**Review by SirusCodes — APPROVED** [review]
LGTM!

File: example/.gitignore (hunk 1 above)
Comment (awhitford) [reply example/.gitignore:56]: Please note that I am simply using the [android gitignore template](https://github.com/toptal/gitignore/blob/master/templates/Android.gitignore), and of course the [Flutter gitignore template](https://github.com/toptal/gitignore/blob/master/templates/Flutter.gitignore).  I _yield_ to the _wisdom_ of the template maintainers.

Honestly, I expected to get away with just `flutter`, but then it showed untracked files:
```
	.idea/
	android/example_android.iml
	example.iml
```
As a result, I reluctantly added the `android` template too.

Comment (SirusCodes) [reply example/.gitignore:56]: Yes, idk why they want to split up stuff. 

This is the .gitignore files for [plugin](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/templates/package/.gitignore.tmpl), [app](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/templates/app/.gitignore.tmpl), [android](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/templates/app/android.tmpl/.gitignore) and [ios](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/templates/app/ios.tmpl/.gitignore)

This is what default you get from flutter create command
What you have done is also fine👍

**Pull request merged**